tensorly/regression/cp_plsr.py

```diff
@@ -122,24 +121,18 @@ def fit(self, X, Y):
             comp_Y_factors_0 = Y[:, 0]
             old_comp_Y_factors_0 = T.ones(T.shape(comp_Y_factors_0)) * T.inf
 
-            # Store the CP result so that we are not starting from scratch
-            Z_comp_CP = "svd"
-
             for iter in range(self.n_iter_max):
                 Z = T.tensordot(X, comp_Y_factors_0, axes=((0,), (0,)))
 
+                if iter == 0:
+                    Z_comp = initialize_cp(Z, 1, normalize_factors=True).factors
+                    Z_comp = [T.reshape(zz, (-1,)) for zz in Z_comp]
+
                 if T.ndim(Z) >= 2:
-                    Z_comp_CP = parafac(
-                        Z,
-                        1,
-                        init=Z_comp_CP,
-                        tol=None,
-                        normalize_factors=False,
-                        n_iter_max=10,
-                    )
-
-                    # Normalize separately—this was profiling very slow in parafac
-                    Z_comp = cp_normalize(Z_comp_CP)[1]
+                    for mode in range(len(Z_comp)):
+                        factor = multi_mode_dot(Z, Z_comp, skip=mode)
```

Comment:
Tucker is equivalent to CP when rank 1. This is more performant, and this spot was consistently one of the slowest parts of testing, particularly for indexing-slow backends.

Reply:
Here, I think this mostly just came down to calling CP, with a lot of extra code contained within, when it is not strictly needed. This call is inside two loops, and so it can be called thousands of times. Within the profiling, I believe most of the time was spent on the MTTKRP, but other parts of the CP call weren't negligible.